components/Boards/PlayerVsPlayer.js

```diff
@@ -0,0 +1,112 @@
+import { Chessboard } from "react-chessboard";
+import Chess from "chess.js";
+import { useState, useEffect } from "react";
+import MoveInfo from "../MoveInfo";
+import PlayerNameDisplay from "../PlayerNameDisplay";
+import GameTerminal from "../GameTerminal";
+
+export default function RandomMoveEngine() {
+  const [game, setGame] = useState(null);
+  const [fen, setFen] = useState(game?.fen());
+  const [previousMove, setPreviousMove] = useState(null);
+  const [moveStatus, setMoveStatus] = useState({});
+  const [history, setHistory] = useState([]);
+
+  const latestHistory = history[history.length - 1];
+
+  //temporarily static player-names
+  const playerName = "Player One";
+  const oppenentName = "Player Two";
+
+  // // ---CREATE GAME OBJECT---
+  useEffect(() => {
+    setGame(new Chess());
+  }, []);
+
+  // // ---CREATE SEPERATE GAME HISTORY TO BYPASS GAME RESETS---
+  function historyStorage() {
+    if (
+      latestHistory?.color !==
+      game?.history({ verbose: true })[
+        game.history({ verbose: true }).length - 1
+      ]?.color
+    ) {
+      setHistory([
+        ...history,
+        game?.history({ verbose: true })[
+          game.history({ verbose: true }).length - 1
+        ],
+      ]);
+    }
+  }
+
+  // // ---CHECK FOR ZOMBIE PIECE => RESPAWN ZOMBIE => RESET GAME OBJECT WITH .fen()---
+  //  ---use previousMove to check captures (flags = "c")
+  useEffect(() => {
+    if (previousMove?.flags === "c") {
+      game.put(
+        { type: previousMove.captured, color: previousMove.color },
+        previousMove.from
+      );
+      const newGame = new Chess(game.fen());
+      setGame(newGame);
+      setFen(newGame.fen());
+      historyStorage();
+    }
+  }, [previousMove]);
+
+  // // ---MAKE A MOVE AND UPDATE GAME OBJECT---
+  function makeAMove(move) {
+    const gameCopy = { ...game };
+    const result = gameCopy.move(move);
+    // setGame(gameCopy);
```

Comment:
you can remove this comment

Reply:
I just left it there for "security" reasons. I commented that line out by mistake, but it still seems to work (still not 100% sure why). So I left it commented out but would rather like to keep it for a bit, if it causes bugs in the future or something.  But I can absolutely delete it, if needed...what do you think?